src/compiler/checker.ts

```diff
@@ -2658,23 +2699,61 @@ module ts {
             }
         }
 
+        function createNumberIndex(value: Type, index?: Type): IndexType {

```

Comment:
"value" is confusing in this context. i apologize if this is too pedantic. but it is hard to follow the change when value/type are used interchangeably  in different context.